src/Agent.Plugins/BuildDropServer.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.BlobStore.Common;
+using Microsoft.VisualStudio.Services.Content.Common.Tracing;
+using Microsoft.VisualStudio.Services.BlobStore.WebApi;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Agent.Sdk;
+using Newtonsoft.Json;
+
+namespace Agent.Plugins.Drop
+{    
+    // A wrapper of BuildDropManager, providing basic functionalities such as uploading and downloading drop artifacts.
+    public class BuildDropClient
+    {
+        // Upload from target path to VSTS BlobStore service through BuildDropManager, then associate it with the build
+        internal async Task UploadDropArtifactAsync(
+            AgentTaskPluginExecutionContext context,
+            VssConnection connection,
+            Guid projectId,
+            int buildId,
+            string name,
+            string source,
+            CancellationToken cancellationToken)
+        {
+            var httpclient = connection.GetClient<DedupStoreHttpClient>();
+            var tracer = new CallbackAppTraceSource(str => context.Output(str), System.Diagnostics.SourceLevels.Information);
+            httpclient.SetTracer(tracer);
+            var client = new DedupStoreClientWithDataport(httpclient, 16 * Environment.ProcessorCount);
+
+            var buildDropManager = new BuildDropManager(client, tracer);
```

Comment:
it seems a little bit weird you have to construct 3 different object before you can construct `BuildDropManager`, i would expect the constructor of `BuildDropManager` take all required args.